packages/discord.js/src/structures/ActionRow.js

```diff
@@ -1,11 +1,21 @@
 'use strict';
 
-const { ActionRow: BuildersActionRow } = require('@discordjs/builders');
-const Transformers = require('../util/Transformers');
+const Component = require('./Component');
+const Components = require('../util/Components');
 
-class ActionRow extends BuildersActionRow {
-  constructor(data) {
-    super(Transformers.toSnakeCase(data));
+/**
+ * Represents an action row
+ * @extends {Component}
+ */
+class ActionRow extends Component {
+  constructor({ components, ...data } = {}) {
+    super(data);
+    /**
+     * The components in this action row
+     * @type {Component[]}
+     * @readonly
+     */
+    this.components = components.map(c => Components.createComponent(c));
```

Comment:
```suggestion
    this.components = components?.map(c => Components.createComponent(c)) ?? [];
```

Reply:
You don't need the `?? []`, right?

Reply:
It would, optional chaining makes it undefined if components is undefined.

Reply:
The constructor does `= {}`, so if no data is passed, this will error.

If explicit data is expected, then remove the default in the constructor.